### ISSUE TITLE
Filter does not work correctly when array contains objects

### ISSUE THREAD

**Issue by Gongreg** [unadlib/mutative#20]
Hello. Thank you for quick replies on previous issues! We've noticed one more issue:

```
import { create } from 'mutative';

const baseState = {
  array: [
    {x: 1}
   ]
};


const state = create(baseState, (draft) => {
  draft.array = draft.array.filter(o => o.x !== 1)
}); 

console.log(state.array) // expected [], received [undefined]

```

This issue seems to happen if filtering is done on array that has objects inside, if it has numbers in it works correctly.

**Comment by unadlib** [comment]
hi @Gongreg ,

I just realized that too.  I've fixed it,  v0.7.1 has been released.

* [fix(draft): fix draft final update value issue](https://github.com/unadlib/mutative/commit/d480a463105cc89f1e1c83bd446e3041824f82cf)
* [fix(test): add testing](https://github.com/unadlib/mutative/commit/bc77b2464a42bd579337af757833eea338112951)

**Comment by Gongreg** [comment]
Wow, blazing fast!

Confirmed that the issue is fixed.
